### PR TITLE
#115 Backup to Google Drive

### PR DESCRIPTION
**Pull request by carlosmiranda**
#115 Added backup to Google Drive functionality to backup script. Also added info to `README.md` about prerequisites, and a puzzle for implementing restore from Google Drive.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#210](https://github.com/timelesslounge/timelessis/issues/210) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#210](https://github.com/timelesslounge/timelessis/issues/210) is assigned to @vladarefiev[/z](https://www.0crat.com/u/vladarefiev), here is [why](https://www.0crat.com/footprint/CF9LG1YCA/a95af61a-b654-4ad0-9609-da177cc30912); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: scripts/backup/pg_backup.sh [hunk 1]
```diff
@@ -82,3 +79,15 @@ else
 fi
 
 echo -e "\nAll database backups complete!"
+
+#####################################
+###### UPLOAD TO GOOGLE DRIVE #######
+#####################################
+
+
+echo -e "\n\\n Uploading backup to Google Drive"
```
Comment (vladarefiev) [inline scripts/backup/pg_backup.sh]: @carlosmiranda why do we need double ‘\’

File: scripts/backup/pg_backup.sh [hunk 2]
```diff
@@ -82,3 +79,15 @@ else
...
+echo -e "--------------------------------------------\n"
+
+gdrive upload $FINAL_BACKUP_DIR"$DATABASE".sql.gz
```
Comment (vladarefiev) [inline scripts/backup/pg_backup.sh]: @carlosmiranda what is about timeouts? Do we need auto retrying for failed uploadings?

**Comment by vladarefiev** [comment]
@carlosmiranda Just a couple comments

File: scripts/backup/pg_backup.sh (hunk 2 above)
Comment (carlosmiranda) [reply scripts/backup/pg_backup.sh]: @vladarefiev please see latest commit. I did not add auto-retry (I think this is out of scope here), but added an error message following the pattern for the other commands in the script.

**Comment by carlosmiranda** [comment]
@vladarefiev please see latest commits.

File: scripts/backup/pg_backup.sh (hunk 2 above)
Comment (vladarefiev) [reply scripts/backup/pg_backup.sh]: @carlosmiranda good, thanks. I would like to add `--timeout 0` too, what do you think?

**Comment by vladarefiev** [comment]
@carlosmiranda one more comment pls

**Comment by carlosmiranda** [comment]
@vladarefiev see latest commit. I added a configurable timeout variable.

**Comment by vladarefiev** [comment]
@carlosmiranda thanks

**Comment by vladarefiev** [comment]
@rultor merge

**Comment by emilianodellacasa** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job was finished in 26 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @vladarefiev[/z](https://www.0crat.com/u/vladarefiev), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#210](https://github.com/timelesslounge/timelessis/issues/210) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa)

**Comment by ypshenychka** [comment]
@vladarefiev According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code. Comments were mostly about design problems, not cosmetic issues

Only two issues were found during code review. Please confirm you'll try to find at least three major issues while future reviews.

**Comment by vladarefiev** [comment]
@ypshenychka correct

**Comment by ypshenychka** [comment]
@vladarefiev thanks

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +20 point(s) just awarded to @vladarefiev[/z](https://www.0crat.com/u/vladarefiev)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**